lib/travis/api/app/endpoint/repos.rb

```diff
@@ -17,7 +17,12 @@ class Repos < Endpoint
       get '/' do
         prefer_follower do
           params['ids'] = params['ids'].split(',') if params['ids'].respond_to?(:split)
-          respond_with service(:find_repos, params)
+          find_repos_service = service(:find_repos, params).run
+          if find_repos_service
+            respond_with find_repos_service
+          else
+            status 404
```

Comment:
you have status 404 here and return 404 below, what's the difference?  If they are the same you could make a method that both of these endpoints call:
``` ruby
def find_repos_responder
  find_repos_service = service(:find_repos, params).run
  return 404 unless find_repos_service
  respond_with find_repos_service
end
```